File: src/ct/game/ai.c
#include <SDL.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <math.h>
#include "ct_stacktrace.h"
#include "ai.h"
#include "engine.h"
#include "utils.h"
#include "ct_array.h"
#include "constants.h"
#include "game.h"

void void_mfunc_AIAbility_clear_AIAbilityPtr(CT_StackTrace* ct_stacktrace, int func_idx_of_call, int line_number_of_call, AIAbility* self) {
    ct_stacktrace_push(ct_stacktrace, func_idx_of_call, line_number_of_call);
    self->is_empty = true;
    ct_stacktrace_pop(ct_stacktrace);
}

void void_mfunc_AIAbility_set_AIAbilityPtr_int_AIAbilityCondition(CT_StackTrace* ct_stacktrace, int func_idx_of_call, int line_number_of_call, AIAbility* self, int unit_ability_handle, AIAbilityCondition condition) {
    ct_stacktrace_push(ct_stacktrace, func_idx_of_call, line_number_of_call);
    self->unit_ability_handle = unit_ability_handle;
    self->condition = condition;
    self->is_empty = false;
    ct_stacktrace_pop(ct_stacktrace);
}

File: src/ct/src/ai.py
from ct.src.engine import SDL_RenderCopy
from ct.src.utils import SDL_Rect, RenderDst, UIEvent, rect_contains_point, set_scaled_rect, set_scaled_rect_with_camera
from ct.src.ct_array import Array
from ct.src.base_types import addr, Uint32, mod, SpriteSrcsPtr
from ct.src.constants import AIAbilityCondition
# weak imports
from ct.src.game import Game


class AIAbility:
    def __init__(self):
        self.unit_ability_handle: int = None
        self.condition: AIAbilityCondition = None
        self.is_empty: bool = None

    def set(self, unit_ability_handle: int, condition: AIAbilityCondition):
        self.unit_ability_handle = unit_ability_handle
        self.condition = condition
        self.is_empty = False

    def clear(self):
        self.is_empty = True
